pkg/nbcontracthelper/helper.go

```diff
@@ -1,9 +1,74 @@
 package nbcontracthelper
 
-const (
-	defaultLinuxUser = "azureuser"
+import (
+	nbcontractv1 "github.com/Azure/agentbaker/pkg/proto/nbcontract/v1"
 )
 
-func GetDefaultLinuxAdminUsername() string {
-	return defaultLinuxUser
+type NBContractBuilder struct {
+	// NBContractConfiguration is the configuration object for the NBContract (Node Bootstrap Contract)
+	nBContractConfiguration *nbcontractv1.Configuration
+}
+
+// Check and initialize each field if it is nil
+func initializeIfNil[T any](field **T) {
+	if *field == nil {
+		*field = new(T)
+	}
+}
+
+// Ensure all objects are non-nil. Please add new objects here.
+func (nBCB *NBContractBuilder) ensureConfigsNonNil() {
+	initializeIfNil(&nBCB.nBContractConfiguration.KubeBinaryConfig)
+	initializeIfNil(&nBCB.nBContractConfiguration.ApiServerConfig)
+	initializeIfNil(&nBCB.nBContractConfiguration.AuthConfig)
+	initializeIfNil(&nBCB.nBContractConfiguration.ClusterConfig)
+	initializeIfNil(&nBCB.nBContractConfiguration.NetworkConfig)
+	initializeIfNil(&nBCB.nBContractConfiguration.GpuConfig)
+	initializeIfNil(&nBCB.nBContractConfiguration.TlsBootstrappingConfig)
+	initializeIfNil(&nBCB.nBContractConfiguration.KubeletConfig)
+	initializeIfNil(&nBCB.nBContractConfiguration.RuncConfig)
+	initializeIfNil(&nBCB.nBContractConfiguration.ContainerdConfig)
+	initializeIfNil(&nBCB.nBContractConfiguration.TeleportConfig)
+	initializeIfNil(&nBCB.nBContractConfiguration.CustomLinuxOsConfig)
+	initializeIfNil(&nBCB.nBContractConfiguration.HttpProxyConfig)
+	initializeIfNil(&nBCB.nBContractConfiguration.CustomCloudConfig)
+	initializeIfNil(&nBCB.nBContractConfiguration.CustomSearchDomainConfig)
+}
+
+// Creates a new instance of NBContractConfig and ensures all objects are non-nil
+func NewNBContractBuilder() *NBContractBuilder {
+	nBCB := &NBContractBuilder{
+		nBContractConfiguration: &nbcontractv1.Configuration{
```

Comment:
can't we just call `ensureConfigsNonNil` here?

Reply:
updated